client/src/lib/dependency-injection/FormValidationManager.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * An API for updating form validation
+ */
+class FormValidationManager {
+  /**
+   * Constructor
+   * @param {object} values
+   */
+  constructor(values) {
+    this.values = values;
+    this.errorMap = {};
+  }
+
+  /**
+   * Adds an error to a given field
+   * @param {string} fieldName
+   * @param {string} message
+   * @returns {FormValidationManager}
+   */
+  addError(fieldName, message) {
+    if (!this.fieldExists(fieldName)) {
+      throw new Error(`Tried to add error to non-existent field: ${fieldName}`);
+    }
+    if (!this.errorMap[fieldName]) {
+      this.errorMap[fieldName] = [];
+    }
+
+    this.errorMap[fieldName] = [
+      ...this.errorMap[fieldName],
+      message
+    ];
+
+    return this;
+  }
+
+  /**
+   * Adds a list of errors to a field
+   * @param {string} fieldName
+   * @param {array} messages
+   * @returns {FormValidationManager}
+   */
+  addErrors(fieldName, messages = []) {
+    messages.forEach(message => this.addError(fieldName, message));
```

Comment:
could we reverse this? to me it's sub-optimal to be calling `addError()` multiple times - instead of calling `addErrors()` once

Reply:
Yeah, not sure what I was thinking here. There's actually not a very good use case for adding multiple errors to one field. I think what I wanted was more of a mapping function:

```
addErrors({
  fieldName: [....],
  otherField: [...]
});
```

Will add that, as it's much more useful.